core/processor/abci.go

```diff
@@ -942,6 +945,65 @@ func (app *App) handleCheckpoint(cpt *types.CheckpointState) error {
 	// this function is called both for interval checkpoints and withdrawal checkpoints
 	event := events.NewCheckpointEvent(app.blockCtx, cpt)
 	app.broker.Send(event)
+	return app.removeOldCheckpoints()
+}
+
+func (app *App) removeOldCheckpoints() error {
+	cpDirPath, err := app.vegaPaths.CreateStatePathFor(paths.StatePath(paths.CheckpointStateHome.String()))
+	if err != nil {
+		return fmt.Errorf("couldn't get checkpoints directory: %w", err)
+	}
+
+	files, err := ioutil.ReadDir(cpDirPath)
+	if err != nil {
+		return fmt.Errorf("could not open open the checkpoint directory: %w", err)
+	}
+
+	// we assume that the files in this directory are only
+	// from the checkpoints
+	// and always keep the last 20, so return if we have less than that
+	if len(files) <= 20 {
```

Comment:
Is there any value in making this an option in the config? Then the system-tests and the snapshot pipeline can set it really low and save reduce the size of the Jenkins artefacts.

Reply:
hum yes I suppose so. I'll do it.

Reply:
addressed